packages/contracts-bedrock/contracts/test/SequencerFeeVault.t.sol

```diff
@@ -73,4 +88,33 @@ contract SequencerFeeVault_Test is Bridge_Initializer {
         // The withdrawal was successful
         assertEq(vault.totalProcessed(), amount);
     }
+
+    function test_withdraw_toL2_succeeds() external {
+        vm.etch(
+            Predeploys.SEQUENCER_FEE_WALLET,
+            address(new SequencerFeeVault(recipient, NON_ZERO_VALUE, FeeVault.WithdrawalNetwork.L2))
+                .code
+        );
+        uint256 amount = vault.MIN_WITHDRAWAL_AMOUNT() + 1;
+        vm.deal(address(vault), amount);
+
+        // No ether has been withdrawn yet
+        assertEq(vault.totalProcessed(), 0);
+
+        vm.expectEmit(true, true, true, true, address(Predeploys.SEQUENCER_FEE_WALLET));
+        emit Withdrawal(
+            address(vault).balance,
+            vault.RECIPIENT(),
+            address(this),
+            FeeVault.WithdrawalNetwork.L2
+        );
+
+        // The entire vault's balance is withdrawn
+        vm.expectCall(recipient, address(vault).balance, bytes(""));
+
+        vault.withdraw();
+
+        // The withdrawal was successful
+        assertEq(vault.totalProcessed(), amount);
```

Comment:
A balance check on the vault having 0 and the recipient having the balance would be good here